scripts/UpdateBuildTools.ps1

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env powershell
+
+<#
+.SYNOPSIS
+    Updates the build tools version and generates a commit message with the list of changes
+.PARAMETER RepoRoot
+    The directory containing the repo
+.PARAMETER GitCommitArgs
+    Additional arguments to pass into git-commit
+.PARAMETER NoCommit
+    Make changes without executing git-commit
+.PARAMETER Force
+    Specified this to make a commit with any changes
+#>
+[cmdletbinding(SupportsShouldProcess = $true)]
+param(
+    [string]$RepoRoot,
+    [string[]]$GitCommitArgs = @(),
+    [switch]$NoCommit,
+    [switch]$Force
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 2
+
+if (-not $RepoRoot) {
+    $RepoRoot = Resolve-Path "$PSScriptRoot\.."
+}
+
+Import-Module "$PSScriptRoot/common.psm1" -Scope Local -Force
+
+function Get-KoreBuildVersion {
+    $lockFile = "$RepoRoot/korebuild-lock.txt"
+    if (!(Test-Path $lockFile)) {
+        return ''
+    }
+    $version = Get-Content $lockFile | Where-Object { $_ -like 'version:*' } | Select-Object -first 1
+    if (!$version) {
+        Write-Error "Failed to parse version from $lockFile. Expected a line that begins with 'version:'"
+    }
+    $version = $version.TrimStart('version:').Trim()
+    return $version
+}
+
+Push-Location $RepoRoot
+try {
+    Ensure-Git
+
+    $oldVersion = Get-KoreBuildVersion
+
+    & "$RepoRoot/run.ps1" -Update default-build '/t:Cow'
+
+    $newVersion = Get-KoreBuildVersion
+
+    if ($oldVersion -eq $newVersion) {
+        Write-Host -ForegroundColor Magenta 'No changes to build tools'
+        exit 0
+    }
+
+    $deps = Get-Content "$RepoRoot/build/dependencies.props" `
+    | % {
+        if ($_ -like '*<InternalAspNetCoreSdkPackageVersion>*') {
+            "    <InternalAspNetCoreSdkPackageVersion>$newVersion</InternalAspNetCoreSdkPackageVersion>"
+        } else {
+            $_
+        }
+    }
+    $deps | Set-Content -Encoding UTF8 "$RepoRoot/build/dependencies.props"
+
+    & git add "$RepoRoot/korebuild-lock.txt"
```

Comment:
Ah, I see, having accepted our fates to have to update `build/dependencies.props` constantly we're also now committing `korebuild-lock.txt`.

Reply:
Yup. #575 requires it.